api/leaderboard.go

```diff
@@ -10,842 +10,761 @@
-			return nil
-		})
-		if err != nil {
-			return FailWithError(err, c)
 		}
-
-		return SucceedWith(serializeMember(member, -1, scoreTTL != ""), c)
+		lg.Debug("Member score increment succeeded.")
+		return nil
+	})
+	if err != nil {
+		return nil, err
 	}
+
+	return newDefaultMemberResponse(member), nil
 }
 
 //RemoveMemberHandler removes a member from a leaderboard
-func RemoveMemberHandler(app *App) func(c echo.Context) error {
-	return func(c echo.Context) error {
-		leaderboardID := c.Param("leaderboardID")
-		memberPublicID := c.Param("memberPublicID")
-		lg := app.Logger.With(
-			zap.String("handler", "RemoveMemberHandler"),
-			zap.String("leaderboard", leaderboardID),
-			zap.String("memberPublicID", memberPublicID),
-		)
+func (app *App) RemoveMember(ctx context.Context, in *api.RemoveMemberRequest) (*api.RemoveMemberResponse, error) {
+	lg := app.Logger.With(
+		zap.String("handler", "RemoveMember"),
+		zap.String("leaderboard", in.LeaderboardId),
+		zap.String("memberPublicID", in.MemberPublicId),
+	)
+
+	err := withSegment("Model", ctx, func() error {
+		lg.Debug("Removing member.")
+		err := app.Leaderboards.RemoveMember(ctx, in.LeaderboardId, in.MemberPublicId)
+
+		if err != nil && !strings.HasPrefix(err.Error(), notFoundError) {
+			lg.Error("Member removal failed.", zap.Error(err))
+			app.AddError()
+			return err
+		}
+		lg.Debug("Member removal succeeded.")
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
 
-		err := WithSegment("Model", c, func() error {
-			lg.Debug("Removing member.")
-			err := app.Leaderboards.RemoveMember(c.StdContext(), leaderboardID, memberPublicID)
+	return &api.RemoveMemberResponse{Success: true}, nil
 
-			if err != nil && !strings.HasPrefix(err.Error(), notFoundError) {
-				lg.Error("Member removal failed.", zap.Error(err))
-				app.AddError()
-				return err
-			}
-			lg.Debug("Member removal succeeded.")
-			return nil
-		})
-		if err != nil {
-			return FailWith(500, err.Error(), c)
-		}
+}
 
-		return SucceedWith(map[string]interface{}{}, c)
+//RemoveMembers removes several members from a leaderboard
+func (app *App) RemoveMembers(ctx context.Context, in *api.RemoveMembersRequest) (*api.RemoveMembersResponse, error) {
+	lg := app.Logger.With(
+		zap.String("handler", "RemoveMembers"),
+		zap.String("leaderboard", in.LeaderboardId),
+	)
+
+	if in.Ids == "" {
+		app.AddError()
+		return nil, status.New(codes.InvalidArgument,
+			"Member IDs are required using the 'ids' querystring parameter").Err()
 	}
-}
 
-//RemoveMembersHandler removes several members from a leaderboard
-func RemoveMembersHandler(app *App) func(c echo.Context) error {
-	return func(c echo.Context) error {
-		leaderboardID := c.Param("leaderboardID")
-		lg := app.Logger.With(
-			zap.String("handler", "RemoveMembersHandler"),
-			zap.String("leaderboard", leaderboardID),
-		)
+	memberIDs := strings.Split(in.Ids, ",")
+	idsInter := make([]interface{}, len(memberIDs))
```

Comment:
This is done this way because of the way the redis library receives information. It receives an interface array. If we change this to a []string, we will have to convert to a []interface anyway.